src/workerd/jsg/modules.c++

```diff
@@ -4,10 +4,38 @@
 
 #include "modules.h"
 #include "promise.h"
+#include <kj/mutex.h>
 
 namespace workerd::jsg {
 namespace {
 
+class CompileCache {
+  // The CompileCache is used to hold cached compilation data for built-in JavaScript modules.
+public:
+  void add(kj::StringPtr key, std::unique_ptr<v8::ScriptCompiler::CachedData> cached) const {
+    cache.lockExclusive()->insert(kj::str(key), kj::mv(cached));
```

Comment:
If two threads try to load the same built-in at the same time, could this insert possibly end up being a duplicate key? If so, this could throw. I think you actually want to just ignore and drop the new value in this case. You can accomplish that with `upsert()` with a no-op callback.

Reply:
Updated to use upsert.